tests/bsim/bluetooth/host/att/timeout/main.c

```diff
@@ -0,0 +1,275 @@
+/* Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <argparse.h>
+#include <zephyr/bluetooth/gatt.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/settings/settings.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/bluetooth/bluetooth.h>
+
+#include "testlib/adv.h"
+#include "testlib/att_read.h"
+#include "testlib/att_write.h"
+#include "bs_macro.h"
+#include "bs_sync.h"
+#include <testlib/conn.h>
+#include "testlib/log_utils.h"
+#include "testlib/scan.h"
+#include "testlib/security.h"
+
+/* This test uses system asserts to fail tests. */
+BUILD_ASSERT(__ASSERT_ON);
+
+LOG_MODULE_REGISTER(main, LOG_LEVEL_DBG);
+
+#define CENTRAL_DEVICE_NBR    0
+#define PERIPHERAL_DEVICE_NBR 1
+
+#define ATT_TIMEOUT_SEC 30
```

Comment:
Arguably we could move `BT_ATT_TIMEOUT` to `att.h` from `att_internal.h` and use that instead of redefining it here